Tiltfile

```diff
@@ -359,7 +359,9 @@ def deploy_worker_templates(template, substitutions):
         calico_values = "./templates/addons/calico/values.yaml"
     flavor_cmd += "; " + helm_cmd + " repo add projectcalico https://projectcalico.docs.tigera.io/charts; " + helm_cmd + " --kubeconfig ./${CLUSTER_NAME}.kubeconfig install calico projectcalico/tigera-operator -f " + calico_values + " --namespace tigera-operator --create-namespace; kubectl --kubeconfig ./${CLUSTER_NAME}.kubeconfig apply -f ./templates/addons/calico/felix-override.yaml"
     if "external-cloud-provider" in flavor_name:
-        flavor_cmd += "; " + helm_cmd + " --kubeconfig ./${CLUSTER_NAME}.kubeconfig install --repo https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/repo cloud-provider-azure --generate-name --set infra.clusterName=${CLUSTER_NAME}"
+        # TODO: remove extra config once the azure-cloud-provider Helm chart release supports VMSS Flex.
+        flavor_cmd += "; " + helm_cmd + " --kubeconfig ./${CLUSTER_NAME}.kubeconfig install --repo https://raw.githubusercontent.com/kubernetes-sigs/cloud-provider-azure/master/helm/repo cloud-provider-azure --generate-name --set infra.clusterName=${CLUSTER_NAME}" + \
+                      " --set cloudControllerManager.imageTag=v1.26.0 --set cloudNodeManager.imageTag=v1.26.0"
```

Comment:
this forces the version of k8s to 1.26 which might not be the case (in fact the tiltfile default for k8s version right now is 1.24 https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2813/files#diff-c2ee8653e1d6b85f0aadf87cd438a9250806c052877248442be4d434cbc52425R24). I think it might make sense to revert this change and then enforce "you must use k8s 1.26+ to enable vmss flex"

Reply:
I added this restriction in documentation, but adding it to validation seems tricky. The natural place to do this is when validating an `AzureMachinePool` and seeing `orchestrationMode: Flexible`. 

But the Kubernetes `version` is on the owner `MachinePool`. I can find it by iterating the owners, but I'm not sure how to fetch the actual resource through the OwnerRef or if that would be appropriate in the validation code.

It would also be ideal if this validation could check for external cloud provider, but that faces similar challenges.

Reply:
Copy/Pasting since the answer is similar: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2965#discussion_r1061046246

Want to open a follow up issue to add validation via webhook?

Reply:
See #2987.